### DISCUSSION TITLE
Save Meta Data for Media Assets

### DISCUSSION THREAD

**Discussion by scottgallant** [tinacms/tinacms#3419] [Ideas]
I'd like to access metadata associated with media assets, such as filesize, file type, extension, dimensions, date created, etc.

**Comment by jamespohalloran** [comment]
Wondering if this metadata be stored in the repo files along with the src url? Or is it just stored under the hood by Tina Cloud?

I'm assuming the GraphQL will respond with an object instead of a string for all `image` requests, so likely a breaking change: 

```diff
- data: {
-      myImg: "blah.png"
- }

+ data: {
+    myImg: {
+           src: "blah.png",
+           format: ""
+.      }
+.  }
+ }
```

**Comment by bensewell** [comment]
Having width & height would be amazing, as this would allow far easier sizing for the next/image component, allowing better optimisation.

**Comment by rufusmai** [comment]
Yes, having a way to retrieve the dimensions of images from the query result would be awesome. 
Especially to prevent layout shifts while the image is being fetched (exactly what `next/image` does).

For example the output could look like:
```js
{
  src: 'foo.img',
  height: 100,
  width: 200,
  format: 'jpeg'
}
```

**Comment by appinteractive** [comment]
Also a blurhash could be saved there that would be passed as the blurDataURL to the next/image or others. That would save a ton of workarounds at build time

**Comment by wspringer** [comment]
I think it makes sense to also have a label here that can be used as alt text.
